### PR TITLE
build: flox output is now just a shell wrapper

### PR DESCRIPTION
**Pull request by garbas**
* possible to set version via env variable at runtime which is then used in flox shell wrapper
* respect `git describe` format
* this should further reduce CI time

### CODE REVIEW

File: cli/flox-rust-sdk/src/flox.rs [hunk 1]
```diff
@@ -33,7 +33,9 @@ use crate::providers::git::GitProvider;
 
 static INPUT_CHARS: Lazy<Vec<char>> = Lazy::new(|| ('a'..='t').collect());
 
-pub const FLOX_VERSION: &str = env!("FLOX_VERSION");
+pub fn get_flox_version() -> String {
+    std::env::var("FLOX_VERSION").unwrap_or(env!("FLOX_VERSION").to_string())
+}
```
Comment (ysndr) [inline cli/flox-rust-sdk/src/flox.rs:38]: We've been using lazy's like this in other places to do runtime overridable inputs

```suggestion
pub static FLOX_VERSION: Lazy<String> = Lazy::new(||
    std::env::var("FLOX_VERSION").unwrap_or(env!("FLOX_VERSION").to_string())
);
```

**Review by ysndr — CHANGES_REQUESTED** [review]
use `Lazy` for runtime overides

File: pkgs/flox/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  version = "${cargoToml.package.version}-${suffix}";
+in
+  symlinkJoin {
+    name = "${flox-cli.pname}-${version}";
+
+    paths = [flox-cli];
+    nativeBuildInputs = [makeWrapper];
+
+    postBuild = ''
+      wrapProgram $out/bin/flox \
```
Comment (tomberek) [inline pkgs/flox/default.nix:32]: possible to use the binary wrapper? `makeBinaryWrapper`

File: pkgs/flox/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        --set PKGDB_BIN       "${flox-pkgdb}/bin/pkgdb" \
+        --set ENV_BUILDER_BIN "${flox-env-builder}/bin/env-builder" \
+        --set FLOX_BIN        "${flox-cli}/bin/flox" \
+        --set FLOX_VERSION    "${version}"
```
Comment (tomberek) [inline pkgs/flox/default.nix:36]: potential problems with subsequent flox calls? `flox activate`, usage during development? Is there a need to unset?

File: pkgs/flox/default.nix (hunk 2 above)
Comment (tomberek) [reply pkgs/flox/default.nix:32]: downsides... hard to see exactly what is inside of it

File: pkgs/flox/default.nix (hunk 3 above)
Comment (garbas) [reply pkgs/flox/default.nix:36]: only if we will use flox to develop flox i think

**Review by tomberek — APPROVED** [review]

**Pull request merged**